app/components/Project/FormBase.tsx

```diff
@@ -0,0 +1,66 @@
+import FormBorder from "components/Layout/FormBorder";
+import { useEffect, useRef } from "react";
+import Form from "lib/theme/service-development-toolkit-form";
+import type { JSONSchema7 } from "json-schema";
+import FormComponentProps from "./FormComponentProps";
+
+interface Props extends FormComponentProps {
+  createInitialData: (data: object) => object;
+  schema: JSONSchema7;
+  uiSchema: object;
+}
+
+const FormBase: React.FunctionComponent<Props> = ({
+  formData,
+  applyChangesFromComponent,
+  onFormErrors,
+  createInitialData,
+  schema,
+  uiSchema,
+}) => {
+  // We restrict the data to only the fields we care about
+  // It's important to default to `undefined` so we trigger the errors on rjsf
```

Comment:
I guess the second part is where [`omitExtraData`](https://react-jsonschema-form.readthedocs.io/en/latest/api-reference/form-props/#omitextradata) doesn't cut it?

Reply:
Yes, `omitExtraData` only works with `onSubmit`, not `onChange` unfortunately